src/averageAges.js

```diff
@@ -15,11 +16,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = century
+    ? people.filter((p) => p.sex === 'm' && Math.ceil(p.died / 100) === century)
+    : people.filter((p) => p.sex === 'm');
+
+  let sum = 0;
+
+  men.forEach((m) => sum += m.died - m.born);
+
+  return sum/men.length;
```

Comment:
Great use of methods to iterate through the array! However, I noticed that the function does not handle cases where the array is empty, which would result in a division by zero. This could lead to unexpected errors or invalid results. I recommend adding a validation to return a default value (such as 0), which would enhance the robustness of the code.